File: Front/neo-store-app/src/assets/scripts/models/task.ts
import { Worker } from './worker';

export class Task {
  id: number;
  name: string;
  description: string;
  difficulty: number;
  isDone: boolean;
  postId: number;
  worker: Worker;

  constructor(
    id: number,
    name: string,
    description: string,
    difficulty: number,
    isDone: boolean,
    postId: number,
    worker: Worker
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.difficulty = difficulty;
    this.isDone = isDone;
    this.postId = postId;
    this.worker = worker;
  }
}

File: src/main/java/ru/bebriki/bebriki/models/Task.java
package ru.bebriki.bebriki.models;

import jakarta.persistence.*;
import lombok.*;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Entity
@Table(name = "tasks")
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "name")
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "difficulty")
    private Integer difficulty;

    @Column(name = "isDone")
    private Boolean isDone;

    @Column(name = "post_id")
    private Integer postId;

    @Column(name = "worker_id")
    private Integer workerId;
}
